stixcore/processing/pipeline.py

```diff
@@ -53,35 +59,145 @@ def __init__(self, func, regex, **args):
+        except Exception as e:
+            logger.error(e)
+            if CONFIG.getboolean('Logging', 'stop_on_error', fallback=False):
+                raise e
+        logger.info(f"GFTSFileHandler:{self.name} end working")
 
     def on_moved(self, event):
+        """Callback if a file was moved
+
+        Parameters
+        ----------
+        event : Event
+            The event object with access to the file
+        """
         if self.regex.match(event.dest_path):
-            self.func(Path(event.dest_path), **self.args)
+            self.queue.put(Path(event.dest_path))
+
+
+class PipelineErrorReport(logging.StreamHandler):
+    """Adds file and mail report Handler to a processing step."""
+    def __init__(self, tm_file):
+        """Create a PipelineErrorReport
+
+        Parameters
+        ----------
+        tm_file : Path
+            The TM file that is process in this step.
+        """
+        logging.StreamHandler.__init__(self)
+
+        self.tm_file = tm_file
+        self.log_dir = Path(CONFIG.get('Pipeline', 'log_dir'))
+        self.log_file = self.log_dir / (tm_file.name + ".log")
+        self.err_file = self.log_dir / (tm_file.name + ".log.err")
+        self.res_file = self.log_dir / (tm_file.name + ".out")
+
+        self.fh = logging.FileHandler(filename=self.log_file, mode="a+")
+        self.fh.setFormatter(logging.Formatter(STX_LOGGER_FORMAT, datefmt=STX_LOGGER_DATE_FORMAT))
+        self.fh.setLevel(logging.getLevelName(CONFIG.get('Pipeline', 'log_level')))
+
+        self.setLevel(logging.ERROR)
+        self.allright = True
+        logging.getLogger().addHandler(self)
+        logging.getLogger().addHandler(self.fh)
+
+    def emit(self, record):
+        """Called in case of a logging event."""
+        self.allright = False
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        logging.getLogger().removeHandler(self)
+        self.fh.flush()
+        logging.getLogger().removeHandler(self.fh)
+        if not self.allright:
+            shutil.copyfile(self.log_file, self.err_file)
+            if CONFIG.getboolean('Pipeline', 'error_mail_send', fallback=False):
+                try:
+                    sender = CONFIG.get('Pipeline', 'error_mail_sender', fallback='localhost')
+                    receivers = CONFIG.get('Pipeline', 'error_mail_receivers').split(",")
+                    host = CONFIG.get('Pipeline', 'error_mail_smpt_host', fallback='localhost')
+                    port = CONFIG.getint('Pipeline', 'error_mail_smpt_port', fallback=25)
+                    smtpObj = smtplib.SMTP(host=host, port=port)
+                    message = f"""Subject: StixCore TMTC Processing Error
+
+Error while processing {self.tm_file}
+
+login to pub099.cs.technik.fhnw.ch and check:
+
+{self.err_file}
+
+StixCore
+
+==========================
+
+do not answer to this mail.
+"""
+                    smtpObj.sendmail(sender, receivers, message)
```

Comment:
For test you could mock this method and just make it has the right contents. Now if you really want to test if the email is sent that won't work but I think that would be an external issue to stixcore